### PR TITLE
Expand "What's New" section

### PR DESCRIPTION
**Pull request by dblevins**
The spirit of this update is to:

 * Slightly expand the detail on changes in 4.0
    * Opted for "javax.ejb" vs "javax" as several javax namespaces referenced in the document are still valid and not renamed.
    * Added short reasons for the removals so that knowledge is in the spec (vs institutional knowledge)
    * Explicitly call out the fact that this specification revision is a breaking change
 * Explicitly add "Jakarta Enterprise Beans 3.2" as this specification is technically part of our history.  Also to call out a potential source of future confusion since it has the same version number as Enterprise JavaBeans 3.2.
 * Use "Enterprise JavaBeans" for the duration of this section where applicable:
    * Legally correct trademark to use for the older specifications
    * Adds a bit of clarity by explicitly noting the name change and history at the JCP
 * Introduce links to EJB 3.1 and 3.0 in Related Documents as these specifications are explicitly referenced.

### CODE REVIEW

**Comment by dblevins** [comment]
@hussainnm Would love your feedback on this one too.  (thanks again for all your contributions)

File: spec/src/main/asciidoc/core/Introduction.adoc [hunk 1]
```diff
@@ -23,35 +23,53 @@ application programmer._
...
+* _Removed the methods relying on JAX-RPC to reflect the removal of
+JAX-RPC from the Jakarta EE 9 Platform._
 
-* _Removed the Support for Distributed Interoperability_
+* _Removed the deprecated EJBContext.getEnvironment() method._
+
+* _Removed the Support for Distributed Interoperability to reflect
+the removal of CORBA from Java 11 and the Jakarta EE 9 Platform._
 
 * _Marked the EJB 2.x API Group as "Optional"_
```
Comment (chengfang) [inline spec/src/main/asciidoc/core/Introduction.adoc:45]: In the current [spec document](https://github.com/eclipse-ee4j/ejb-api/blob/master/spec/src/main/asciidoc/core/RuntimeEnvironment.adoc), the term "Enterprise Beans 2.x API" is used. I think we should use the new term name.

File: spec/src/main/asciidoc/core/Introduction.adoc [hunk 2]
```diff
@@ -23,35 +23,53 @@ application programmer._
...
 === What is New in This Release
 
-* _The 'javax' namespace/packages have been renamed to 'jakarta'_
+The Enterprise Beans 4.0 specification is a breaking change from
+prior Enterprise Beans specifications delivering a namespace change
+from `javax` to `jakarta` across the API along with removal of
+deprecated APIs.
 
-* _Removed the methods relying on java.security.Identity_
+* _All 'javax.ejb' namespace/packages have been renamed to 'jakarta.ejb'_
```
Comment (chengfang) [inline spec/src/main/asciidoc/core/Introduction.adoc:31]: My personal preference of formatting, either `javax.ejb` `jakarta.ejb`, or the plain javax.ejb jakarta.ejb. The single quote seems a bit odd here.

Comment (dblevins) [reply spec/src/main/asciidoc/core/Introduction.adoc]: Good catch.  Fixed.

File: spec/src/main/asciidoc/core/Introduction.adoc (hunk 1 above)
Comment (dblevins) [reply spec/src/main/asciidoc/core/Introduction.adoc:45]: Note that isn't a change part of this PR.  That said, the trademark rules here are a little nuanced.  Here's the discussions we've had in the Specification/Steering committees and the guidance we've been given on use of either trademark.

 - When referring to the specification in general or referring to a new version of the specification, we should use the new name.
 - When explicitly referring to past JCP versions of the specification we should take care to use the appropriate trademark.  For example,  "J2EE 1.2" not "Jakarta EE 1.2"

Similarly, when "Java EE" became the new name for "J2EE" we continued to refer to the older specs using their proper name "J2EE 1.3", "J2EE 1.4", etc and did not being referring to them as "Java EE 1.3",  "Java EE 1.4", etc.

Comment (hussainnm) [reply spec/src/main/asciidoc/core/Introduction.adoc:45]: +1 to change EJB 2.x to Enterprise Beans 2.x.

As I have explained in the mailing list, an earlier version under the new namespace should be referred with the new name.

Comment (tkburroughs) [reply spec/src/main/asciidoc/core/Introduction.adoc:45]: +1 to change EJB 2.x to Enterprise Beans 2.x.

This does not appear to be a reference to an older spec level, but instead is a reference to a subset of the current specification, specifically the 2.x API Group, so should use the same terminology as in the rest of the current version of the spec.

**Pull request merged**